Source/WebInspectorUI/UserInterface/Views/DOMTreeContentView.js

```diff
@@ -694,6 +702,97 @@ WI.DOMTreeContentView = class DOMTreeContentView extends WI.ContentView
         this._showPrintStylesChanged();
     }
 
+    _handleOverrideUserPreferencesButtonClicked()
+    {
+        if (this._userPreferencesOverridesPopover) {
+            this._userPreferencesOverridesPopover.dismiss();
+            this._userPreferencesOverridesPopover = null;
+            return;
+        }
+
+        let contentElement = document.createElement("div");
+        contentElement.classList.add("user-preferences-content");
+
+        let createSelectElement = (label, preference, values = [], defaultValue) => {
+            if (!preference || !values.length)
+                return;
+
+            let labelElement = contentElement.appendChild(document.createElement("label"));
+            let labelTextElement = labelElement.appendChild(document.createElement("span"));
+            labelTextElement.textContent = label;
+
+            let selectElement = labelElement.appendChild(document.createElement("select"));
+            selectElement.setAttribute("name", preference);
+
+            for (let value of values) {
+                let optionElement = selectElement.appendChild(document.createElement("option"));
+                optionElement.value = value;
+                optionElement.textContent = this._userPreferenceValueLabel(value);
+
+                let selectedValue = this._overridenUserPreferences.get(preference) || defaultValue;
+
+                if (value === selectedValue)
+                    optionElement.setAttribute("selected", true);
+            }
+
+            selectElement.addEventListener("change", (event) => {
+                let value = event.target.value;
+
+                WI.cssManager.overrideUserPreference(preference, value);
+
+                if (value !== defaultValue)
+                    this._overridenUserPreferences.set(preference, value);
+                else
+                    this._overridenUserPreferences.delete(preference);
+
+                this._overrideUserPreferencesNavigationItem.activated = this._overridenUserPreferences.size > 0;
+            });
+
+            return selectElement;
+        }
+
+        let calculateTargetFrame = () => {
+            return WI.Rect.rectFromClientRect(this._overrideUserPreferencesNavigationItem.element.getBoundingClientRect()).pad(2);
+        }
+
+        let preferredEdges = [WI.RectEdge.MAX_Y, WI.RectEdge.MAX_X];
+
+        this._userPreferencesOverridesPopover = new WI.Popover(this);
+        this._userPreferencesOverridesPopover.windowResizeHandler = () => {
```

Comment:
Aside: we should probably audit all `set windowResizeHandler` to make sure that we're not keeping things alive longer than they should be (e.g. `this` is being kept alive here)